### PR TITLE
bug fixes and enhancements

### PR DESCRIPTION
**Pull request by Shubhankar-Cashfree**
- Add common exception handler for all status code.
- Pass the raw response in case of non 2xx with x-request-id.
- Fix the request field for bulk transfer api (accept vpa etc.)
- Remove get transfer list api.
- Added pubicKey authentication

### CODE REVIEW

File: src/main/java/com/cashfree/lib/experiments/PayoutExperiments.java [hunk 1]
```diff
@@ -42,13 +42,26 @@
   }
   
   public static void main(String[] args) {
+
+    String clientId = "CF27JBMB8GN4CHAQI6Q";
```
[the remark this answers is not in the document]
Comment (Shubhankar-Cashfree) [reply src/main/java/com/cashfree/lib/experiments/PayoutExperiments.java]: Removed

File: src/main/java/com/cashfree/lib/http/UriBuilder.java [hunk 2]
```diff
@@ -23,6 +25,7 @@ public UriBuilder queryParam(String paramName, String paramValue) {
     if (paramsMap == null) {
       paramsMap = new HashMap<>();
     }
+    paramValue = paramValue.replace(" ", "%20");
```
[the remark this answers is not in the document]
Comment (Shubhankar-Cashfree) [reply src/main/java/com/cashfree/lib/http/UriBuilder.java]: Deleted the file, as we no longer need it

File: src/main/java/com/cashfree/lib/payout/domains/response/CfPayoutsResponse.java [hunk 3]
```diff
@@ -11,4 +11,6 @@
   private Integer subCode;
 
   private String message;
+
+  private String requestId;
```
[the remark this answers is not in the document]
Comment (Shubhankar-Cashfree) [reply src/main/java/com/cashfree/lib/payout/domains/response/CfPayoutsResponse.java]: Changed.

File: src/main/java/com/cashfree/lib/payout/domains/request/BatchTransferRequest.java [hunk 4]
```diff
@@ -22,11 +22,15 @@
 
   private List<Payload> batch;
 
+  private String paymentInstrumentId;
```
[the remark this answers is not in the document]
Comment (Shubhankar-Cashfree) [reply src/main/java/com/cashfree/lib/payout/domains/request/BatchTransferRequest.java]: Removed

File: src/main/java/com/cashfree/lib/payout/constants/PayoutConstants.java [hunk 5]
```diff
@@ -25,7 +25,7 @@
 
   public static final String GET_TRANSFER_STATUS_REL_URL = PAYOUT_REL_URL + VERSION + "/getTransferStatus";
 
-  public static final String GET_TRANSFERS_REL_URL = PAYOUT_REL_URL + VERSION + "/getTransfers";
+//  public static final String GET_TRANSFERS_REL_URL = PAYOUT_REL_URL + VERSION + "/getTransfers";
```
[the remark this answers is not in the document]
Comment (Shubhankar-Cashfree) [reply src/main/java/com/cashfree/lib/payout/constants/PayoutConstants.java]: Done

File: src/main/java/com/cashfree/lib/payout/clients/Beneficiary.java [hunk 6]
```diff
@@ -20,61 +21,102 @@ public Beneficiary(Payouts payouts) {
   }
 
   public boolean addBeneficiary(BeneficiaryDetails beneficiary) {
+    URIBuilder uriBuilder = null;
+    try {
+      uriBuilder =
+              new URIBuilder(Payouts.getEndpoint() + PayoutConstants.ADD_BENEFICIARY_REL_URL);
+    } catch (Exception e) {
+      e.printStackTrace();
```
[the remark this answers is not in the document]
Comment (Shubhankar-Cashfree) [reply src/main/java/com/cashfree/lib/payout/clients/Beneficiary.java]: threw exception

File: README.md [hunk 7]
```diff
@@ -30,7 +30,9 @@ Init the package with your credentials and add the below code in your config.py
 import com.cashfree.lib.payout.clients.Payouts;
 
 // Get instance for Cashfree Payout
-Payouts payouts = Payouts.getInstance(Environment.PRODUCTION, "client_id", "client_secret");
+client_id = ""; //your client id here
```
Comment (ankur-cashfree) [inline README.md]: String clientID ;

**Review by ankur-cashfree — APPROVED** [review]
LGTM

**Pull request merged**